src/course-unit/course-sequence/hooks.js

```diff
@@ -12,15 +12,15 @@ export function useSequenceNavigationMetadata(courseId, currentSequenceId, curre
   const isLastUnit = !nextUrl;
   const sequenceIds = useSelector(getSequenceIds);
   const sequenceIndex = sequenceIds.indexOf(currentSequenceId);
-  const unitIndex = sequence.unitIds.indexOf(currentUnitId);
+  const unitIndex = sequence.unitIds?.indexOf(currentUnitId);
```

Comment:
Adding `?` means that `unitIndex` can be `undefined`, and in that case we shouldn't be adding to it below on line 21. If you convert this to TypeScript it would show this as an error:

![Screenshot showing error](https://github.com/user-attachments/assets/d43b5b39-822c-4a84-89fe-fab68d0410aa)